### PR TITLE
More refactoring

### PR DESCRIPTION
**Pull request by iliketigers**

### CODE REVIEW

File: src/PortfolioSummary.test.js [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import * as Utilities from './utils/utilities';
 import * as DateUtils from '../utils/dateUtils';
 
-const today = DateUtils.getTodaysDateInIso().split('T')[0];
+const today = new Date().toISOString().split('T')[0];
```
Comment (kamahen) [inline src/PortfolioSummary.test.js:4]: `const [today, _] = newDate().toISOString(.split('T');`
Also,  this gets you the date in UTC - is that what you want?

File: src/components/PortfolioChart.js [hunk 2]
```diff
@@ -22,7 +22,7 @@ export class PortfolioChart extends Component {
         this.state.endDate._d
       );
     } else {
-      const today = DateUtils.getToday();
+      const today = DateUtils.convertToPickedDate(new Date());
       let startDate = new Date();
       startDate.setHours(12, 0, 0, 0);
```
Comment (kamahen) [inline src/components/PortfolioChart.js:27]: You have a converToPickedDate (which should be renamed) -- why aren't you using it here?

File: src/components/PortfolioSummary.js [hunk 3]
```diff
@@ -18,7 +18,7 @@ class PortfolioSummary extends Component {
     if (!portfolio) {
       return 0;
     }
-    const today = DateUtils.getToday();
+    const today = DateUtils.convertToPickedDate(new Date());
     currValue = await ChartUtils.getYAxisValue(portfolio, today);
     return currValue;
```
Comment (kamahen) [inline src/components/PortfolioSummary.js:23]: Pretty sure you can just do:
```
return await ChartUtils.getYAxisValue(portfolio, today);
```

File: src/constants.js [hunk 4]
```diff
@@ -1,7 +1,7 @@
-import * as DateUtils from './utils/dateUtils';
-
-const today = DateUtils.getTodaysDateInIso();
-const yesterday = DateUtils.getYesterdaysDateInIso();
+const today = new Date().toISOString();
+let yesterday = new Date();
+yesterday.setDate(yesterday.getDate() - 1);
```
Comment (kamahen) [inline src/constants.js:3]: Does `yesterday.getDate()` - 1 do what you expect on the 1st of the month?
Better to subtract 24 hours, I would think.

File: src/utils/apiCalls.js [hunk 5]
```diff
@@ -14,7 +14,10 @@ export function getTodaysPrice(ticker) {
 }
 
 export function getYesterdaysPrice(ticker) {
```
Comment (kamahen) [inline src/utils/apiCalls.js:16]: Does this function belong in dateUtils? Seems to introduce ticker stuff, which is hardly date-related.

File: src/utils/dateUtils.js [hunk 6]
```diff
@@ -1,10 +1,6 @@
...
-  const today = new Date();
-  today.setHours(12, 0, 0, 0);
-  return today;
-};
-
 /**
+ * Takes two dates and creates an array of all dates
+ * between them, inclusive.
  *
  * @param {Object} startDate
```
Comment (kamahen) [inline src/utils/dateUtils.js:5]: Is `Object` correct? - shouldn't it be `Date`?

File: src/utils/dateUtils.js [hunk 7]
```diff
@@ -20,6 +16,18 @@ export const getDateRange = (startDate, endDate) => {
...
 };
 
+/**
+ * Converts Date objects to match the date picker's
+ * format. Basically the date picker just defaults
+ * dates to local date, 12pm.
+ *
+ * @param {Object} date
+ */
+export const convertToPickedDate = (date) => {
```
Comment (kamahen) [inline src/utils/dateUtils.js:26]: convertToPickedDate - maybe: setToBeginningOfDay

Also, your convertToPickedDate() modifies the argument -- this is really bad style. This is better:
```
export const convertToPickedDate = (date) => {
  let beginning_of_day = new Date(date);
  beginning_of_day.setHours(12, 0, 0, 0);
  return beginning_of_day;
};
```
(There might be a better way of doing this; my JS-fu is not good.)

File: src/utils/dateUtils.js [hunk 8]
```diff
@@ -36,15 +44,3 @@ export const convertPickedDateToUtc = (date) => {
   newDate.setHours(newDate.getHours() - timezoneOffset / 60 - 12);
```
Comment (kamahen) [inline src/utils/dateUtils.js:44]: Does this work properly when the current time is earlier than the timezoneOffset? (e.g., if timezoneOffset is 8 hours, does this do what you want when the time is 7am?)

In general, it's best to do arithmetic on date or time objects (add/subtract days/hours/minutes/whatever) and then convert to printable.

Comment (kamahen) [reply src/utils/dateUtils.js:44]: To be more precise:
```
d = new Date();  // gets current time
e = new Date(d.getTime() + 1);  // gets current time + 1 second
```
Doing arithmetic this way deals with wrap-around problems - e.g. `d.setHours(-1)` doesn't do what you expect.

File: src/PortfolioSummary.test.js (hunk 1 above)
Comment (iliketigers) [reply src/PortfolioSummary.test.js:4]: yes

File: src/components/PortfolioChart.js (hunk 2 above)
Comment (iliketigers) [reply src/components/PortfolioChart.js:27]: ? i am using it?

File: src/constants.js (hunk 4 above)
Comment (iliketigers) [reply src/constants.js:3]: yes

File: src/components/PortfolioSummary.js (hunk 3 above)
Comment (iliketigers) [reply src/components/PortfolioSummary.js:23]: done

File: src/utils/dateUtils.js (hunk 6 above)
Comment (iliketigers) [reply src/utils/dateUtils.js:5]: sure

File: src/PortfolioSummary.test.js (hunk 1 above)
Comment (iliketigers) [reply src/PortfolioSummary.test.js:4]: - wait why the [today, _] ?? how is that better?

File: src/utils/dateUtils.js (hunk 7 above)
Comment (iliketigers) [reply src/utils/dateUtils.js:26]: done, still thinking about the function name tho, i might end up changing the functionality...

File: src/utils/dateUtils.js (hunk 8 above)
Comment (iliketigers) [reply src/utils/dateUtils.js:44]: what do you mean? I am doing it that way... date.setHours(date.getHours - x)

haven't thoroughly tested this yet but seems to work; i need to update/create tests. i was getting overwhelmed with trying to figure out tests, so i was just focusing on getting the dang thing to work first.

File: src/utils/apiCalls.js (hunk 5 above)
Comment (iliketigers) [reply src/utils/apiCalls.js:16]: this is in apiCalls, not dateUtils.

**Pull request merged**